examples/voting/contract/src/Voting.sol

```diff
@@ -2,157 +2,126 @@
 pragma solidity ^0.8.28;
 
 import {requireTimeBefore, requireTimeAfter} from "pod-sdk/Time.sol";
-import {requireQuorum} from "pod-sdk/Quorum.sol";
+import {FastTypes} from "pod-sdk/FastTypes.sol";
 
 contract Voting {
+    using FastTypes for FastTypes.SharedCounter;
+    using FastTypes for FastTypes.OwnedCounter;
+
     enum VoterState {
         Unregistered,
         Registered,
         Voted
     }
 
-    struct Poll {
+    struct Proposal {
         uint256 deadline;
-        uint256 maxChoice;
-        address owner;
+	uint256 threshold;
+        address proposer;
         mapping(address => VoterState) voters;
-        mapping(uint256 => uint256) voteCount;
         uint256 totalVotes;
         uint256 totalVoters; // Total number of registered voters
-        uint256 winningChoice; // 0 means not set
+        bool executed;
+        bytes data;
     }
 
-    // Maps poll ID to poll data
-    mapping(bytes32 => Poll) private polls;
-
-    event PollCreated(bytes32 indexed pollId, uint256 deadline);
-    event Voted(bytes32 indexed pollId, address indexed voter, uint256 indexed choice);
-    event Winner(bytes32 indexed pollId, uint256 indexed choice);
-
-    /// @notice Calculate poll ID
-    /// @param deadline The poll deadline in seconds
-    /// @param maxChoice The maximum choice in a poll
-    /// @param owner The crator of the poll
-    /// @param voters The poll participants
-    /// @return pollId The unique poll ID derived from the input parameters
-    function getPollId(uint256 deadline, uint256 maxChoice, address owner, address[] calldata voters)
+    // Maps proposal ID to proposal data
+    mapping(bytes32 => Proposal) private proposals;
+    FastTypes.SharedCounter private voteCount;
+    FastTypes.OwnedCounter private hasVoted;
```

Comment:
What's the benefit of using these fast types compared to the previous mappings inside the `Proposal` struct? There seems to be a downside of more memory usage (duplication of proposal_id key).